File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int> targetmap;

        for(int i=0;i<n;i++)
        {
            int diff = target-nums[i];
            if(targetmap.find(diff)!=targetmap.end())
            {
                return {targetmap[diff],i};
            }
            targetmap.insert({nums[i],i});
        }
        return {};




        
    }
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    prevMap := make(map[int]int)

    for i,n := range nums {
        diff := target - n
        if j,found := prevMap[diff];found {
            return []int{j,i}
        }
        prevMap[n]=i
    }
    return []int{}
    
}